src/validate.c

```diff
@@ -109,38 +110,57 @@ int main(int argc,char **argv)	{
 
     char *filename = argv[1]; 
 
-    unsigned long skip_offset = 0;
-    unsigned long skip_length = 0;
+    unsigned long skip_offset_sig = 0;
+    unsigned long skip_length_sig = 0;
+
+    unsigned long skip_offset_key = 0;
+    unsigned long skip_length_key = 0;
   
-    if (!appimage_get_elf_section_offset_and_length(filename, ".sha256_sig", &skip_offset, &skip_length)) {
+    if (!appimage_get_elf_section_offset_and_length(filename, ".sha256_sig", &skip_offset_sig, &skip_length_sig)) {
         fprintf(stderr, "Failed to read .sha256_sig section");
         exit(1);
     }
+    if (!appimage_get_elf_section_offset_and_length(filename, ".sig_key", &skip_offset_key, &skip_length_key)) {
+        fprintf(stderr, "Failed to read .sig_key section");
+        exit(1);
+    }
 
-    if(skip_length > 0) {
-        fprintf(stderr, "Skipping ELF section %s with offset %lu, length %lu\n", segment_name, skip_offset, skip_length);
+    if(skip_length_sig > 0) {
+        fprintf(stderr, "Skipping ELF section %s with offset %lu, length %lu\n", segment_name, skip_offset_sig, skip_length_sig);
     } else {
         fprintf(stderr, "ELF section %s not found, is the file signed?\n", segment_name);
         exit(1);
     }
+    if(skip_length_key > 0) {
+        fprintf(stderr, "Skipping ELF section %s with offset %lu, length %lu\n", segment_key_name, skip_offset_key, skip_length_key);
+    } else {
+        fprintf(stderr, "ELF section %s not found, is the file signed?\n", segment_key_name);
+        exit(1);
+    }
+    if(skip_offset_sig + skip_length_sig != skip_offset_key) {
+      fprintf(stderr, "Validate only worlds when .sha256_sig and .sig_key are next to one another in the ELF\n");
```

Comment:
Typo: "works". But I'd like to suggest a different wording.

```suggestion
      fprintf(stderr, "validate only works when .sha256_sig and .sig_key are contiguous in the ELF header\n");
```

Reply:
How about "next to each other in the ELF file"? I would probably not immediately know what "contiguous" means in this context.

Reply:
I updated the wording.